### PR TITLE
Embedding api key with request header property

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Embedding api key with request header property

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -45,19 +45,19 @@ stargate:
...
+          enabled: true
           url: https://api.openai.com/v1/
           api-key: YOUR_API_KEY
 
         # Hugging face embedding service configuration
         huggingface:
-          enabled: false
-          url: https://api-inference.huggingface.co
+          enabled: true
+          url: https://api-inference.huggingface.co/pipeline/feature-extraction
```
Comment (maheshrajamani) [inline src/main/resources/application.yaml]: Not sure on what is the default url for hugging-face. Anyway property that can be override but need to be fixed.

File: src/test/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/UpdateManyCommandResolverTest.java [hunk 2]
```diff
@@ -221,75 +219,6 @@ public void dynamicFilterCondition() throws Exception {
               });
     }
 
-    @Test
```
Comment (maheshrajamani) [inline src/test/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/UpdateManyCommandResolverTest.java:224]: This test was moved into CommandResolverWithVectorizerTest

File: src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/test/CustomITEmbeddingService.java [hunk 3]
```diff
@@ -29,11 +32,13 @@ public class CustomITEmbeddingService implements EmbeddingService {
...
     TEST_DATA.put("Updating new data", new float[] {0.22f, 0.55f, 0.68f, 0.36f, 0.6f});
   }
-  ;
 
   @Override
-  public Uni<List<float[]>> vectorize(List<String> texts) {
+  public Uni<List<float[]>> vectorize(List<String> texts, Optional<String> apiKeyOverride) {
     List<float[]> response = new ArrayList<>(texts.size());
+    if (texts.size() == 0) return Uni.createFrom().item(response);
+    if (!apiKeyOverride.isPresent() || !apiKeyOverride.get().equals(TEST_API_KEY))
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/test/CustomITEmbeddingService.java:40]: To test header key for vectorization IT

File: src/main/java/io/stargate/sgv2/jsonapi/api/request/DataApiRequestInfo.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import io.stargate.sgv2.api.common.token.CassandraTokenResolver;
+import io.vertx.ext.web.RoutingContext;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.enterprise.inject.Instance;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.SecurityContext;
+import java.util.Optional;
+
+/**
+ * This class is used to get the request info like tenantId, cassandraToken and embeddingApiKey This
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/request/DataApiRequestInfo.java:13]: nit: missing `.` between embeddingApiKey, This.

**Review by tatu-at-datastax — APPROVED** [review]
LGTM (esp. if this can be split in 2 as discussed)

File: src/main/java/io/stargate/sgv2/jsonapi/api/request/DataApiRequestInfo.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+public class DataApiRequestInfo {
+  private final Optional<String> embeddingApiKey;
+  private final Optional<String> tenantId;
+  private final Optional<String> cassandraToken;
+
+  @Inject
+  public DataApiRequestInfo(
+      RoutingContext routingContext,
+      SecurityContext securityContext,
+      Instance<TenantResolver> tenantResolver,
```
[the remark this answers is not in the document]
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/api/request/DataApiRequestInfo.java:26]: These are from quarkus-commons where there are multiple implementation. Configuration drives which instance to be wired.

**Pull request merged**